xcode/Subconscious/Shared/Components/Editor/EditorModalSheetView.swift

```diff
@@ -0,0 +1,265 @@
+//
+//  EditorModalSheetView.swift
+//  Subconscious
+//
+//  Created by Ben Follington on 4/3/2024.
+//
+
+import SwiftUI
+import ObservableStore
+import os
+
+enum EditorModalSheetAction: Equatable, Hashable {
+    case editEntry(EntryStub)
+    case postPublicly
+    case dismiss
+    case setPresented(Bool)
+}
+
+extension AppAction {
+    static func from(_ notification: MemoEditorDetailNotification) -> Self? {
+        switch notification {
+        case let .requestSaveEntry(entry):
+            return .saveEntry(entry)
+        case let .requestDelete(address):
+            return .deleteEntry(address)
+        case let .requestMoveEntry(from, to):
+            return .moveEntry(from: from, to: to)
+        case let .requestMergeEntry(parent, child):
+            return .mergeEntry(parent: parent, child: child)
+        case let .requestUpdateAudience(address, audience):
+            return .updateAudience(address: address, audience: audience)
+        case let .requestAssignNoteColor(address, color):
+            return .assignColor(address: address, color: color)
+        default:
+            return nil
+        }
+    }
+}
+
+struct EditorModalSheetModel: ModelProtocol, Equatable {
```

Comment:
This is a very thin model, I imagine it will merge with the main editor model in the future.